### ISSUE TITLE
input-requires-label rule should allow input tag nested inside label tag

### ISSUE THREAD

**Issue by at6ue** [htmlhint/HTMLHint#1170]
Currently, the input-requires-label rule forces the input tag to have an id attribute to associate a label tag, but the input tag without id attribute can be considered to be associated with a nested label tag.

> Alternatively, you can nest the &lt;input&gt; directly inside the &lt;label&gt;, in which case the for and id attributes are not needed because the association is implicit:
[&lt;label&gt;: The Input Label element - HTML: HyperText Markup Language | MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#:~:text=Alternatively%2C%20you%20can%20nest)

According to [HTML Standard](https://html.spec.whatwg.org/multipage/forms.html#the-label-element:~:text=If%20the%20for%20attribute%20is%20not%20specified%2C%20but%20the%20label%20element%20has%20a%20labelable%20element%20descendant%2C%20then%20the%20first%20such%20descendant%20in%20tree%20order%20is%20the%20label%20element%27s%20labeled%20control.), 
> If the for attribute is not specified, but the label element has a labelable element descendant, then the first such descendant in tree order is the label element's labeled control.

So, my idea is to modify the rule to allow the first input tag nested inside a label tag without the for attribute.

Here is a small example: clicking on the "bar" label toggles the "baz" checkbox associated with the for attribute, instead of the nested "bar" checkbox.
https://jsfiddle.net/pL5s0mvk/

**Comment by nschonni** [comment]
Yes, the wrapping is valid, so you don't need to turn on this rule if you're using that style. All rules are opt-in

**Comment by nneto** [comment]
This rule is also making htmlhint complain about type="submit" inputs.
